### DISCUSSION TITLE
I am using the runner scaleset (i.e., controller plus scalesets with their own values.yaml files).  Is HPA an option?

### DISCUSSION THREAD

**Discussion by dashap** [actions/actions-runner-controller#3051] [Questions]
Hello,

Is using replicas and setting cpu limit and requests plus HPA an option for when you use the setup that has the controller and scalesets (dind)?  

I cannot seem to find info/directions on how to allow the pods to have replicas?  I want to set limit/request values for cpu low, but allow the pods to scale based on cpu 50% or greater for example.  The commands/documents like https://learn.microsoft.com/en-us/azure/aks/tutorial-kubernetes-scale?tabs=azure-cli#autoscale-pods don't seem to apply for github runners using arc and runner scalesets.  The only deployment I see is for the controller itself.  The controller values.yaml does allow you to set replica count greater than one, but I think that would just apply to the controller, not the runners themselves.  

Please let me know if HPA is an option or not or if I am missing the point somehow of the controller/runner scaleset solution and why.

**Comment by nikola-jokic** [comment]
Hey,

You are correct, deployment is only used to create controller manager deployment. The way scale-sets are created is that they are spinning up runners based on the workflows acquired from the service. There is no HPA in the ARC running scale sets. It also would not work. We are creating a JIT token for a runner, so each runner has only one registration token. Spinning up the same pod would result in a duplicated session and fail. HPA is not an option :relaxed:

**Reply by dashap** [reply]
Thank you for the response

**Reply by rositaj91** [reply]
Hi,

Sorry to open this discussion again. The ARC Running Scale sets that you mentioned here are the listeners for each runner group right. So this means having two listeners/replica for the same runner group is not possible.

I guess this answers the question here as well right? https://github.com/orgs/community/discussions/160487
